contrib/pkg/awsprivatelink/enable.go

```diff
@@ -0,0 +1,219 @@
+				Values: []*string{aws.String(o.infraId + "-vpc")},
+			},
+		},
+	})
+	if err != nil {
+		log.WithError(err).Fatal("Failed to get VPC of the active cluster")
+	}
+	if len(describeVPCsOutput.Vpcs) == 0 {
+		log.Fatal("VPC of the active cluster not found")
+	}
+	vpcID := *describeVPCsOutput.Vpcs[0].VpcId
+	log.Debugf("Found VPC ID = %v for the active cluster", vpcID)
+
+	// Create Hub account credentials Secret
+	hiveNS := hiveConfig.Spec.TargetNamespace
+	if hiveNS == "" {
+		hiveNS = constants.DefaultHiveNamespace
+	}
+	hubCredentialsSecret, err := o.generateAWSCredentialsSecret(hiveNS)
+	if err != nil {
+		log.WithError(err).Fatal("Failed to generate Secret with AWS credentials")
+	}
+	if err = o.dynamicClient.Create(context.Background(), hubCredentialsSecret); err != nil {
+		log.WithError(err).Fatal("Failed to create Secret with hub account credentials")
+	}
+	log.Info("Secret with hub account credentials created")
+
+	// Update HiveConfig
+	hiveConfig.Spec.AWSPrivateLink = &hivev1.AWSPrivateLinkConfig{}
+	hiveConfig.Spec.AWSPrivateLink.AssociatedVPCs = append(
+		hiveConfig.Spec.AWSPrivateLink.AssociatedVPCs,
+		hivev1.AWSAssociatedVPC{
+			AWSPrivateLinkVPC: hivev1.AWSPrivateLinkVPC{VPCID: vpcID, Region: o.region},
+		},
+	)
+	hiveConfig.Spec.AWSPrivateLink.CredentialsSecretRef = corev1.LocalObjectReference{Name: hubCredentialsSecret.Name}
+	hiveConfig.Spec.AWSPrivateLink.DNSRecordType = hivev1.AWSPrivateLinkDNSRecordType(o.dnsRecordType)
+	if err = o.dynamicClient.Update(context.Background(), hiveConfig); err != nil {
+		log.WithError(err).Fatal("Failed to update HiveConfig")
+	}
+	log.Info("HiveConfig updated")
+
+	return nil
+}
+
+func (o *enableOptions) getRegionAndInfraId() (region string, infraId string, err error) {
+	infrastructure := &configv1.Infrastructure{}
+	if err = o.dynamicClient.Get(context.Background(), types.NamespacedName{Name: "cluster"}, infrastructure); err != nil {
+		return
+	}
+	if infrastructure.Status.PlatformStatus == nil {
+		err = errors.New("Infrastructure.status.platformStatus is empty")
+		return
+	}
+	if infrastructure.Status.PlatformStatus.AWS == nil {
+		err = errors.New("Infrastructure.status.platformStatus.aws is empty")
+		return
+	}
+
+	region = infrastructure.Status.PlatformStatus.AWS.Region
+	infraId = infrastructure.Status.InfrastructureName
+	return
+}
+
+// Adapted from contrib/pkg/adm/managedns/enable.go:generateAWSCredentialsSecret()
+func (o *enableOptions) generateAWSCredentialsSecret(namespace string) (*corev1.Secret, error) {
+	defaultCredsFilePath := filepath.Join(o.homeDir, ".aws", "credentials")
+	accessKeyID, secretAccessKey, err := awsutils.GetAWSCreds("", defaultCredsFilePath)
+	if err != nil {
+		return nil, err
+	}
+	return &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("awsprivatelink-hub-acct-creds-%s", uuid.New().String()[:5]),
```

Comment:
Okay, so I think this is the crux of the "multiple secrets" issue. Why shouldn't we just have the name hardcoded and the same every time? That would make a few things easier:
- We would not risk proliferation of redundant secrets across failed `enable` attempts.
- We could do without the labeling. (Though I guess we could add the label as an extra layer of safety for the deletion, so in the case where the secret already existed before `enable`, we wouldn't delete it via `disable`.)
- If the secret `Create()` errors with `IsAlreadyExists`, we could proceed rather than failing. This would give us the added flexibility of using an existing secret -- possibly created by someone with more powerful credentials.

Thoughts?

Reply:
Sound great. I'm going with what you suggested here. 

For disable, only secrets with the label will be deleted. So the ones configured manually will persist.